assets/src/blocks/CarouselHeader/Sidebar.js

```diff
@@ -3,50 +3,117 @@ import {
   PanelBody,
   PanelRow,
 } from '@wordpress/components';
+import { Fragment } from '@wordpress/element';
 
 import { InspectorControls } from '@wordpress/block-editor';
 import { SidebarSlide } from './SidebarSlide';
+import { useState, useRef } from 'react';
 
 const { __ } = wp.i18n;
 
+const reOrderSlides = (slides, oldIndex, newIndex) => {
+  const movingSlide = slides[oldIndex];
+  return slides.reduce((orderedSlides, slide, index) => {
+    if (index === newIndex) {
+      orderedSlides.push(movingSlide);
+    }
+    if (index !== oldIndex) {
+      orderedSlides.push(slide);
+    }
+    if (index === slides.length -1 && newIndex > index) {
+      orderedSlides.push(movingSlide);
+    }
+    return orderedSlides;
+  }, []);
+}
+
 export const Sidebar = ({
   carouselAutoplay,
   setAttributes,
   slides,
   currentSlide,
   changeSlideAttribute,
   goToSlide,
-}) => (
-  <InspectorControls>
-    <PanelBody title={__('Setting', 'planet4-blocks-backend')}>
+}) => {
+  const [dragTarget, setDragTarget] = useState(null);
+  const [draggedSlide, setDraggedSlide] = useState(null);
+  const timeoutRef = useRef(null);
+
+  return <InspectorControls>
+    <PanelBody title={ __('Setting', 'planet4-blocks-backend') }>
       <CheckboxControl
-        label={__('Carousel Autoplay', 'planet4-blocks-backend')}
-        help={__('Select to trigger images autoslide', 'planet4-blocks-backend')}
-        value={carouselAutoplay}
-        checked={carouselAutoplay === true}
-        onChange={value => setAttributes({ carousel_autoplay: value })}
+        label={ __('Carousel Autoplay', 'planet4-blocks-backend') }
+        help={ __('Select to trigger images autoslide', 'planet4-blocks-backend') }
+        value={ carouselAutoplay }
+        checked={ carouselAutoplay === true }
+        onChange={ value => setAttributes({ carousel_autoplay: value }) }
       />
     </PanelBody>
-    {slides.map((slide, index) => (
+    { slides.map((slide, index) => <div
+      style={ {
+        transform: index === draggedSlide ? 'scaleY(0)' : 'none',
```

Comment:
When you start dragging an element, what is shown while dragging is a bitmap snapshot of what the element looked like the moment the drag started. The `.01s` delay should be plenty to ensure that snapshot is taken before we hide the element.